### PR TITLE
Ipr

### PR DESCRIPTION
**Pull request by haggi**
Move renderview update from tilecallback back to renderqueueworker to update it in the main thread instead from another thread.

### CODE REVIEW

**Comment by dictoon** [comment]
Does that resolves the Render View update problems you were seeing?

**Comment by haggi** [comment]
Yes, renderview updates in normal render and in IPR now. Except the last bucket borders.

**Comment by dictoon** [comment]
> Except the last bucket borders

What do you mean?

File: src/renderqueueworker.cpp [hunk 1]
```diff
@@ -76,6 +76,48 @@ concurrent_queue<Event>* gEventQueue()
...
+                    bottom != yMin ||
+                    top != yMax)
+                    return;
+            }
+        }
+
+        MRenderView::updatePixels(xMin, xMax, yMin, yMax, pixels);
+        MRenderView::refresh(xMin, xMax, yMin, yMax);
+    }
+}
```
Comment (dictoon) [inline src/renderqueueworker.cpp]: You can collapse the two anonymous namespaces into one.

File: src/renderqueueworker.cpp [hunk 2]
```diff
@@ -707,6 +749,10 @@ void RenderQueueWorker::startRenderQueueWorker()
           case Event::ADDIPRCALLBACKS:
             addIPRCallbacks();
             break;
+
+          case Event::UPDATEUI:
+              updateRenderView(e.xMin, e.xMax, e.yMin, e.yMax, e.pixels.get());
```
Comment (dictoon) [inline src/renderqueueworker.cpp]: Fix indentation (see code just above).

**Comment by haggi** [comment]
The same problem you discovered. The "doNotClearBackground" option in MRenderView::startRender(). I'll set it to true what should remove the outline.

**Comment by dictoon** [comment]
Looks good, thanks!

**Pull request merged**